src/bin/pg_autoctl/keeper_config.c

```diff
@@ -814,3 +822,87 @@ keeper_config_update_with_absolute_pgdata(KeeperConfig *config)
 	}
 	return true;
 }
+
+
+/*
+ * keeper_config_validate_replication_password gates the update of the replication
+ * password in the config file if the server is a standby and it is running. If
+ * the server is not running, validation will be performed at server restart. If
+ * the server is a primary, the ensuing reload request will lead to the node
+ * active process updating the password in the database.
+ */
+bool
+keeper_config_validate_replication_password(KeeperConfig *config, const char *password)
```

Comment:
This looks to me like a modularity violation. The `keeper_config.c` file is supposed to deal with the configuration file and data structure, not with the actual Postgres instance that we're managing. That code belongs to `primary_standby.c` on a first reading of it, with maybe a wrapper to call it from `keeper.c`.